bundle/config/mutator/python/apply_python_mutator_test.go

```diff
@@ -0,0 +1,450 @@
+package python
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"reflect"
+	"runtime"
+	"testing"
+
+	"golang.org/x/exp/maps"
+
+	"github.com/databricks/cli/libs/dyn"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/bundle/config"
+	assert "github.com/databricks/cli/libs/dyn/dynassert"
+	"github.com/databricks/cli/libs/process"
+)
+
+func TestApplyPythonMutator_Name_load(t *testing.T) {
+	mutator := ApplyPythonMutator(ApplyPythonMutatorPhaseLoad)
+
+	assert.Equal(t, "ApplyPythonMutator(load)", mutator.Name())
+}
+
+func TestApplyPythonMutator_Name_init(t *testing.T) {
+	mutator := ApplyPythonMutator(ApplyPythonMutatorPhaseInit)
+
+	assert.Equal(t, "ApplyPythonMutator(init)", mutator.Name())
+}
+
+func TestApplyPythonMutator_load(t *testing.T) {
+	withFakeVEnv(t, ".venv")
+
+	b := loadYaml("databricks.yml", `
+      experimental:
+        pydabs:
+          enabled: true
+          venv_path: .venv
+      resources:
+        jobs:
+          job0:
+            name: job_0`)
+
+	ctx := withProcessStub(
+		[]string{
+			interpreterPath(".venv"),
+			"-m",
+			"databricks.bundles.build",
+			"--phase",
+			"load",
+		},
+		`{
+			"experimental": {
+				"pydabs": {
+					"enabled": true,
+					"venv_path": ".venv"
+				}
+			},
+			"resources": {
+				"jobs": {
+					"job0": {
+						name: "job_0"
+					},
+					"job1": {
+						name: "job_1"
+					},
+				}
+			}
+		}`)
+
+	mutator := ApplyPythonMutator(ApplyPythonMutatorPhaseLoad)
+	diag := bundle.Apply(ctx, b, mutator)
+
+	assert.NoError(t, diag.Error())
+
+	assert.ElementsMatch(t, []string{"job0", "job1"}, maps.Keys(b.Config.Resources.Jobs))
+
+	if job0, ok := b.Config.Resources.Jobs["job0"]; ok {
+		assert.Equal(t, "job_0", job0.Name)
+	}
+
+	if job1, ok := b.Config.Resources.Jobs["job1"]; ok {
+		assert.Equal(t, "job_1", job1.Name)
+	}
```

Comment:
Potential for follow-up: assert that the location information is persisted across the call.

As is, this would pass if the mutator uses only the process output.

Reply:
Agree. There are tests for it in dyn.Override, but since this is important behaviour, we should have a simple test for ApplyPythonMutator as well.

Reply:
I will add in a separate PR